sdk/core/Azure.Core/src/Internal/GenericOperationSource.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Primitives;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Core
+{
+    internal class GenericOperationSource<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors | DynamicallyAccessedMemberTypes.NonPublicConstructors)] T> : IOperationSource<T> where T : IPersistableModel<T>
+    {
+        T IOperationSource<T>.CreateResult(Response response, CancellationToken cancellationToken)
+            => CreateResult(response);
+
+        ValueTask<T> IOperationSource<T>.CreateResultAsync(Response response, CancellationToken cancellationToken)
+            => new ValueTask<T>(CreateResult(response));
+
+        private T CreateResult(Response response)
+            => (T)ModelReaderWriter.Read(response.Content, typeof(T))!;
```

Comment:
When using the null-forgiving operator, `!`, are we 100% sure that MRW.Read can never return null for this case?

Reply:
Updated this to avoid cast.

If we take a look at the [ModelReaderWriter.Read<T>](https://github.com/Azure/azure-sdk-for-net/blob/325e3d543938f7465451f4840042be2f03fc182b/sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriter.cs#L97-L104), it throws at null data input and the below return statement is `T` instead of `T?`, I am not sure why the method returns `T?` in the first place.
```csharp
return GetInstance<T>().Create(data, options);
```